browser-switch/src/main/java/com/braintreepayments/api/BrowserSwitchPersistentStore.java

```diff
@@ -48,7 +51,35 @@ void putActiveRequest(BrowserSwitchRequest request, Context context) {
         }
     }
 
+    void putActiveResult(BrowserSwitchResult result, Context context) {
+        try {
+            PersistentStore.put(RESULT_KEY, result.toJson(), context);
+        } catch (JSONException e) {
+            Log.d(TAG, e.getMessage());
+            Log.d(TAG, Arrays.toString(e.getStackTrace()));
+        }
+    }
+
+    BrowserSwitchResult getActiveResult(Context context) {
+        BrowserSwitchResult request = null;
+
+        String activeResultJSON = PersistentStore.get(RESULT_KEY, context);
+        if (activeResultJSON != null) {
+            try {
+                request = BrowserSwitchResult.fromJson(activeResultJSON);
+            } catch (JSONException e) {
+                Log.d(TAG, e.getMessage());
+                Log.d(TAG, Arrays.toString(e.getStackTrace()));
```

Comment:
Would we rather return an error to the merchant instead of log?

Reply:
Ideally yeah, the messed up part is it's been done this way historically for `putActiveRequest` too lol.

We do have an opportunity to make the code more clear in a next major version bump and reporting persistent storage errors may be one of the things we address down the line? There are also some alternatives to persistent storage that I've seen if we end up porting this library to Kotlin.